lib/connection.js

```diff
@@ -227,7 +227,8 @@ C.open = function(allFields, openCallback0) {
       tunedOptions.channelMax =
         negotiate(fields.channelMax, allFields.channelMax);
       tunedOptions.heartbeat =
-        negotiate(fields.heartbeat, allFields.heartbeat);
+        allFields.heartbeat === 0 ?
```

Comment:
Can this ternary expression just be simplified to

```js
tunedOptions = allFields.heartbeat === null ? fields.heartbeat : allFields.heartbeat
```

i.e., as discussed, if given in options use that value, otherwise use the server value?

Reply:
ok, fixed